objects/src/accounts/package/storage_entry/mod.rs

```diff
@@ -0,0 +1,606 @@
+        }
+    }
+
+    /// Returns the name of the storage entry.
+    pub fn name(&self) -> &str {
+        match self {
+            StorageEntry::Value { name, .. } => name.as_str(),
+            StorageEntry::Map { name, .. } => name.as_str(),
+            StorageEntry::MultiSlot { name, .. } => name.as_str(),
+        }
+    }
+
+    /// Returns the optional description of the storage entry.
+    pub fn description(&self) -> Option<&str> {
+        match self {
+            StorageEntry::Value { description, .. } => description.as_deref(),
+            StorageEntry::Map { description, .. } => description.as_deref(),
+            StorageEntry::MultiSlot { description, .. } => description.as_deref(),
+        }
+    }
+
+    /// Returns all the `WordRepresentation` values covered by this entry.
+    /// For `Value` entries, this returns a single-element slice.
+    /// For `MultiSlot` entries, this returns all values.
+    /// For `Map` entries, since they're key-value pairs, return an empty slice.
+    pub fn word_values(&self) -> &[WordRepresentation] {
+        match self {
+            StorageEntry::Value { value, .. } => std::slice::from_ref(value),
+            StorageEntry::MultiSlot { values, .. } => values.as_slice(),
+            StorageEntry::Map { .. } => &[],
+        }
+    }
+
+    /// Returns the map entries for a `Map` variant as a slice.
+    /// Returns an empty slice for non-map variants.
+    pub fn map_entries(&self) -> &[MapEntry] {
+        match self {
+            StorageEntry::Map { values, .. } => values.as_slice(),
+            StorageEntry::Value { .. } => &[],
+            StorageEntry::MultiSlot { .. } => &[],
+        }
+    }
+
+    /// Returns an iterator over all of the storage entries's template keys.
+    // TODO: Should template keys be typed?
+    pub fn template_keys(&self) -> Box<dyn Iterator<Item = &TemplateKey> + '_> {
+        match self {
+            StorageEntry::Value { value, .. } => value.template_keys(),
+            StorageEntry::Map { values, .. } => {
+                Box::new(values.iter().flat_map(|word| word.template_keys()))
+            },
+            StorageEntry::MultiSlot { values, .. } => {
+                Box::new(values.iter().flat_map(|word| word.template_keys()))
+            },
+        }
+    }
+
+    /// Attempts to convert the storage entry into a list of [StorageSlot].
+    ///
+    /// - StorageEntry::Value would convert to a [StorageSlot::Value]
+    /// - StorageEntry::MultiSlot would convert to as many [StorageSlot::Value] as defined
+    /// - StorageEntry::Map would convert to a [StorageSlot::Map]
+    ///
+    /// Each of the entry's values could be dynamic. These values are replaced for values found
+    /// in `template_values`, identified by its key.
+    pub fn try_into_storage_slots(
+        self,
+        template_values: &BTreeMap<String, TemplateValue>,
+    ) -> Result<Vec<StorageSlot>, ComponentPackageError> {
```

Comment:
Could this take `&self` rather than `self` as a parameter? (seems like we usually clone `StorageEntry` before calling `try_into_storage_slot()` on them.

Reply:
Changed this. Thought clippy might complain for a wrong self convention, but it was not the case.

Reply:
Maybe a better name here would be `try_build_storage_slots()`?